ls1intum/Artemis
# `Development`: Unify Json usage

### Checklist
#### General


#### Server

### Motivation and Context


Currently, different JSON implementations are used.

### Description

Unified Json implementations to use gson library. Also some of the JsonParser implementations are replaced with ObjectMapper. Added an architectural test to prevent the use of other user libraries in the future.

### Steps for Testing

Prerequisites:
- Please use test server 11 where LTI is enabled

1. Log in to https://moodle.ase.in.tum.de/ using username: student1 and password: as same password with artemis_test_user_1 
2. Navigate to LTI Testing course and select TS11 Authentication Test Exercise.
3. Login to Artemis using artemis_test_user_1  credentials.
4. Verify you can see the exercise and participate in the exercise.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage






no change in test coverage since only replaced Json implementations.

### Screenshots
